pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2390,9 +2390,10 @@ type NetworkSpec struct {
 type NetworkProviderType string
```

Comment:
There is an enum above this that looks like this:

```golang
// +kubebuilder:validation:Enum="";host;multus
```

This enum is the list of allowed Provider entries in CRDs. It's part of the CRD generation stuff provided by kubebuilder. This will need to be modified to add the new mode. You will likely find that the CRD will error if you try to define `spec.network.provider: host-strict` in the YAML manifest without this change.

```golang
// +kubebuilder:validation:Enum="";host;host-strict;multus
```

Reply:
I added `host-strict` to this Enum and ran `make crds` and `make codegen` but did not see any errors locally.